ndcube/ndcube_sequence.py

```diff
@@ -239,6 +244,72 @@ def explode_along_axis(self, axis):
         # creating a new sequence with the result_cubes keeping the meta and common axis as axis
         return self._new_instance(result_cubes, common_axis=new_common_axis, meta=self.meta)
 
+    def _get_sequence_axes_wcs(self):
+        """
+        Returns a WCS object corresponding to the sequence axis of this NDCubeSequence.
+        It can contain multiple pixel axes that map to the same world axis.
+        """
+        table_coords = []
+
+        for axis_name, axis_coords in self.sequence_axis_coords.items():
+            if isinstance(axis_coords[0], u.Quantity):
+                table_coords.append(QuantityTableCoordinate(u.Quantity(axis_coords),
+                                                            names=axis_name))
+
+            elif isinstance(axis_coords[0], Time):
+                table_coords.append(TimeTableCoordinate(Time(axis_coords),
+                                                        names=axis_name))
+
+            elif isinstance(axis_coords[0], SkyCoordTableCoordinate):
+                table_coords.append(SkyCoordTableCoordinate(SkyCoord(axis_coords),
+                                                            names=axis_name))
+
+            else:
+                raise Exception(f"Unable to convert global_coords '{axis_name}' into an axis. "
+                                "global_coords must be either `astropy.units.Quantity`, "
+                                "`astropy.time.Time`, or `astropy.coordinates.SkyCoord`.")
+
+        return MultipleTableCoordinate(*table_coords).wcs
+
+    def combine_cubes(self, common_wcs_index=0):
+        """
+        Reprojects all `~ndcube.NDCube` objects to a common WCS, and stacks the data together
+        to return a single (N+1)-dimensional `~ndcube.NDCube` with an associated Compound WCS
+        containing extra axes corresponding to the sequence axis.
+
+        Parameters
+        ----------
+        common_wcs_index: `int`, optional
+            The index of the `~ndcube.NDCube` whose WCS is to be used as the base on which to
+            reproject all other `~ndcube.NDCube`s in the `~ndcube.NDCubeSequence`.
+
+        Returns
+        -------
+        `ndcube.NDCube`
+            A combined (N+1)D cube with stacked data of all cubes.
+        """
+        from ndcube import NDCube  # adding this here to prevent a circular import
+
```

Comment:
Insert a check here to ensure all cubes have the same shape.  For now, if cubes have different shapes, raise a `ValueError`.

(`NDCubeSequence` supports some limited cases were cubes are of different shapes.)